src/components/birth-date-picker/birth-date-picker.tsx

```diff
@@ -0,0 +1,172 @@
+import React = require("react");
+import {autorun, computed, observable, reaction} from "mobx";
+import {observer} from "mobx-react";
+
+function isValidDate(date: any): boolean {
+    return date instanceof Date && !Number.isNaN(date.getTime());
+}
+
+function yearMonthDayFromDate(date: Date | undefined) {
+    const valid = isValidDate(date);
+    return [
+        valid ? String(date!.getUTCFullYear()) : "",
+        valid ? String(date!.getUTCMonth() + 1) : "",
+        valid ? String(date!.getUTCDate()) : ""
```

Comment:
This seems like a waste, why do you need to check valid 3 times? you can try something like:
if (!isValidDate(...)) {
  return ['','','']
} 
....